### DISCUSSION TITLE
Can we expose `openMenu` in the `Menu` component easily?

### DISCUSSION THREAD

**Discussion by bobmulder** [tailwindlabs/headlessui#2912] [Ideas]
Hi guys!

I find myself in a situation where I want to open the menu from headlessui programatically. This seems not to be possible at the moment. I decided to dig into the code and found the following line at https://github.com/tailwindlabs/headlessui/blob/main/packages/%40headlessui-vue/src/components/menu/menu.ts#L239:

```ts
    return () => {
      let slot = { open: menuState.value === MenuStates.Open, close: api.closeMenu }
      return render({ ourProps: {}, theirProps: props, slot, slots, attrs, name: 'Menu' })
    }
```

I'd say that adding `openMenu` to the slot variable would expose that method to the slot data, so we can use it in our own templates, right? 

Code would look like:

```ts
    return () => {
      let slot = { open: menuState.value === MenuStates.Open, close: api.closeMenu, openMenu: api.openMenu }
      return render({ ourProps: {}, theirProps: props, slot, slots, attrs, name: 'Menu' })
    }
```

Can someone confirm this would be a welcome addition to the package, or is there any reason this function is not exposed?

Unfortunately my knowledge about contributing to js packages is limited. If someone is able to give some support I'm willing to help wherever possible.

Regards, Bob

**Comment by emkr** [comment]
Yeah, running into a situation where I would need this as well
